src/Container.php

```diff
@@ -64,13 +72,14 @@ public function add($alias, $concrete = null, $singleton = false)
             return null;
         }
 
-        // get a definition of the item
-        $this->items[$alias]['singleton'] = (boolean) $singleton;
-
+        // we need to build a definition
         $factory    = $this->getDefinitionFactory();
         $definition = $factory($alias, $concrete, $this);
 
-        $this->items[$alias]['definition'] = $definition;
+        $this->items[$alias] = [
+            'definition' => $definition,
+            'singleton'  => (boolean) $singleton
```

Comment:
hm, does it really matter? I prefer bool as well, but isn't it a personal preference? Are there any standards for it?